eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/CDDL.hs

```diff
@@ -0,0 +1,1008 @@
+      =:= mp
+        [ opt $ idx 0 ==> arr [0 <+ a int64] -- Plutus v1, only 166 integers are used, but more are accepted (and ignored)
+        , opt $ idx 1 ==> arr [0 <+ a int64] -- Plutus v2, only 175 integers are used, but more are accepted (and ignored)
+        , opt $ idx 2 ==> arr [0 <+ a int64] -- Plutus v3, only 223 integers are used, but more are accepted (and ignored)
+        , 0 <+ asKey (3 ... 255) ==> arr [0 <+ a int64] -- Any 8-bit unsigned number can be used as a key.
+        ]
+
+transaction_metadatum :: Rule
+transaction_metadatum =
+  "transaction_metadatum"
+    =:= smp [0 <+ asKey transaction_metadatum ==> transaction_metadatum]
+    // sarr [0 <+ a transaction_metadatum]
+    // VInt
+    // (VBytes `sized` (0 :: Word64, 64 :: Word64))
+    // (VText `sized` (0 :: Word64, 64 :: Word64))
+
+transaction_metadatum_label :: Rule
+transaction_metadatum_label = "transaction_metadatum_label" =:= (VUInt `sized` (8 :: Word64))
+
+metadata :: Rule
+metadata =
+  "metadata"
+    =:= mp
+      [ 0
+          <+ asKey transaction_metadatum_label
+          ==> transaction_metadatum
+      ]
+
+auxiliary_data :: Rule
+auxiliary_data =
+  "auxiliary_data"
+    =:= metadata -- Shelley
+    // sarr
+      [ "transaction_metadata" ==> metadata -- Shelley-ma
+      , "auxiliary_scripts" ==> arr [0 <+ a native_script]
+      ]
+    // tag
+      259
+      ( mp
+          [ opt (idx 0 ==> metadata) -- Alonzo and beyond
+          , opt (idx 1 ==> arr [0 <+ a native_script])
+          , opt (idx 2 ==> arr [0 <+ a plutus_v1_script])
+          , opt (idx 3 ==> arr [0 <+ a plutus_v2_script])
+          , opt (idx 4 ==> arr [0 <+ a plutus_v3_script])
+          ]
+      )
+
+vkeywitness :: Rule
+vkeywitness = "vkeywitness" =:= arr [a vkey, a signature]
+
+bootstrap_witness :: Rule
+bootstrap_witness =
+  "bootstrap_witness"
+    =:= arr
+      [ "public_key" ==> vkey
+      , "signature" ==> signature
+      , "chain_code" ==> (VBytes `sized` (32 :: Word64))
+      , "attributes" ==> VBytes
+      ]
+
+native_script :: Rule
+native_script =
+  "native_script"
+    =:= arr
+      [ a script_pubkey
+          / script_all
+          / script_any
+          / script_n_of_k
+          / invalid_before
+          -- Timelock validity intervals are half-open intervals [a, b).
+          -- This field specifies the left (included) endpoint a.
+          / invalid_hereafter
+          -- Timelock validity intervals are half-open intervals [a, b).
+          -- This field specifies the right (excluded) endpoint b.
+      ]
+
+script_pubkey :: Named Group
+script_pubkey = "script_pubkey" =:~ grp [0, a addr_keyhash]
+
+script_all :: Named Group
+script_all = "script_all" =:~ grp [1, a (arr [0 <+ a native_script])]
+
+script_any :: Named Group
+script_any = "script_any" =:~ grp [2, a (arr [0 <+ a native_script])]
+
+script_n_of_k :: Named Group
+script_n_of_k =
+  "script_n_of_k"
+    =:~ grp [3, "n" ==> VUInt, a (arr [0 <+ a native_script])]
```

Comment:
This is actually incorrect in the original spec. It is a signed 64bit `Int`. Could you please fix it here and in the spec. It's fine if you'd prefer not to change that in this PR then just feel free to convert this comment into a ticket and we'll take care of it later
```suggestion
    =:~ grp [3, "n" ==> int64, a (arr [0 <+ a native_script])]
```